tests/ui/auxiliary/external_macro.rs

```diff
@@ -0,0 +1,15 @@
+#[macro_export]
+macro_rules! external_struct_rest_default {
+    () => {
+        #[derive(Default)]
+        struct ExternalDefault {
+            a: i32,
+            b: i32,
+        }
+
+        let _ = ExternalDefault {
+            a: 10,
+            ..Default::default()
+        };
+    };
+}
```

Comment:
Can't you use `proc_macros::external!{ … }` in your tests, without defining a dedicated external macro?